pkg/extensions/sync/remote.go

```diff
@@ -50,13 +51,33 @@ func (registry *RemoteRegistry) GetContext() *types.SystemContext {
 func (registry *RemoteRegistry) GetRepositories(ctx context.Context) ([]string, error) {
 	var catalog catalog
 
-	_, _, _, err := registry.client.MakeGetRequest(ctx, &catalog, "application/json", //nolint: dogsled
+	_, header, _, err := registry.client.MakeGetRequest(ctx, &catalog, "application/json", "", //nolint: dogsled
 		constants.RoutePrefix, constants.ExtCatalogPrefix)
 	if err != nil {
 		return []string{}, err
 	}
 
-	return catalog.Repositories, nil
+	repos := catalog.Repositories
```

Comment:
I think the problem of duplicating the 2nd page in place of the 1st page in the final list starts here. Further down `catalog` is passed down as a reference as well and the `Repositories` property is overwritten with the 2nd page result, thus `repos` get "overwritten" as well. Further iterations in the loop work, because at `append(...)` the list capacity need to expand and a new list is reserved in memory after that.

This way the behaviour is flaky. Its just a matter of luck to some extent when the list gets expanded and which page get lost or overwritten.

Reply:
Note that the list can be inconsistent in general since repos may be get added/deleted/updated in between pagination results.